internal/service/cognitoidentity/openid_token_for_developer_identity_ephemeral.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cognitoidentity
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/YakDriver/regexache"
+	"github.com/aws/aws-sdk-go-v2/service/cognitoidentity"
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/mapvalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral"
+	"github.com/hashicorp/terraform-plugin-framework/ephemeral/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework"
+	"github.com/hashicorp/terraform-provider-aws/internal/framework/flex"
+	fwtypes "github.com/hashicorp/terraform-provider-aws/internal/framework/types"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+// @EphemeralResource("aws_cognito_identity_openid_token_for_developer_identity", name="Cognito Identity Open ID Token for Developer Identity")
+func newEphemeralCognitoIdentityOpenIDTokenForDeveloperIdentity(context.Context) (ephemeral.EphemeralResourceWithConfigure, error) {
+	return &ephemeralCognitoIdentityOpenIDTokenForDeveloperIdentity{}, nil
+}
+
+const (
+	EPNameCognitoIdentityOpenIDToken = "Cognito Identity Open ID Token for Developer Identity Ephemeral Resource"
+)
+
+type ephemeralCognitoIdentityOpenIDTokenForDeveloperIdentity struct {
+	framework.EphemeralResourceWithConfigure
+}
+
+func (e *ephemeralCognitoIdentityOpenIDTokenForDeveloperIdentity) Metadata(_ context.Context, req ephemeral.MetadataRequest, resp *ephemeral.MetadataResponse) {
+	resp.TypeName = "aws_cognito_identity_openid_token_for_developer_identity"
+}
+
+func (e *ephemeralCognitoIdentityOpenIDTokenForDeveloperIdentity) Schema(ctx context.Context, req ephemeral.SchemaRequest, resp *ephemeral.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"identity_id": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
```

Comment:
Open Question if this should be required or not. There is potential for a [rate limiting](https://github.com/hashicorp/terraform-provider-aws/issues/29673#issuecomment-1775743457), but there's not existing resources to generate the identity id.

Reply:
Let's keep as-is to match the AWS API.